perf/collector_libpfm.go

```
@@ -136,14 +152,27 @@ func (c *collector) setup() error {
 	c.cpuFilesLock.Lock()
 	defer c.cpuFilesLock.Unlock()
 	cgroupFd := int(cgroup.Fd())
-	for _, group := range c.events.Core.Events {
-		customEvent, ok := c.eventToCustomEvent[group[0]]
-		var err error
-		if ok {
-			err = c.setupRawNonGrouped(customEvent, cgroupFd)
-		} else {
-			err = c.setupNonGrouped(string(group[0]), cgroupFd)
+	for i, group := range c.events.Core.Events {
+		// CPUs file descriptors of group leader needed for perf_event_open.
+		leaderFileDescriptors := new([]int)
+		*leaderFileDescriptors = make([]int, c.numCores)
+		for j, event := range group.events {
+			// First element is group leader.
+			groupLeader := j == 0
+			customEvent, ok := c.eventToCustomEvent[event]
+			if ok {
+				err = c.setupRawEvent(customEvent, cgroupFd, i, leaderFileDescriptors, groupLeader)
+			} else {
+				err = c.setupEvent(string(event), cgroupFd, i, leaderFileDescriptors, groupLeader)
+			}
+		}
+
+		// Group is prepared so we should reset and enable counting.
+		for _, fd := range *leaderFileDescriptors {
+			err = unix.IoctlSetInt(fd, unix.PERF_EVENT_IOC_RESET, 0)
```

this error is not checked, please handle this error